CONTRIBUTING.md

```diff
@@ -1,341 +1,180 @@
-   and reboot.
-
-4. Use the docker daemon inside minikube for building:
-   ```bash
-   eval $(minikube docker-env)
-   ```
+1. Clone your fork to your local computer.
 
-5. Clone the zero-to-jupyterhub repo:
    ```bash
-   git clone git@github.com:jupyterhub/zero-to-jupyterhub-k8s.git
+   git clone http://github.com/<YOUR-GITHUB-USERNAME>/zero-to-jupyterhub-k8s.git
    cd zero-to-jupyterhub-k8s
-   ```
 
-6. Create a virtualenv & install the libraries required for builds to happen:
-   ```bash
-   python3 -m venv .
-   source bin/activate
-   python3 -m pip install -r dev-requirements.txt
+   # make it easy to reference the projects GitHub repository as "upstream"
+   git remote add upstream https://github.com/jupyterhub/zero-to-jupyterhub-k8s
+
+   # make it obvious what you reference by renaming a reference to your
+   # personal GitHub repository to "fork"
+   git remote rename origin fork
    ```
 
-7. Now run `chartpress` to build the requisite docker images inside minikube:
-    ```bash
-    chartpress
-    ```
+### Prepare Virtual Machine software
 
-    This will build the docker images inside minikube & modify
-    `jupyterhub/values.yaml` with the appropriate values to make the chart
-    installable!
+A `Vagrantfile` is a way to prepare a Virtual Machine (VM), and we [have
+one](Vagrantfile) to prepare a VM for local development! We can use it to get
+a VM up and running, enter it with SSH, develop and run tests, and later shut
+down without influencing our system.
 
-8. Configure helm and minikube for RBAC:
-   ```bash
-   kubectl create clusterrolebinding add-on-cluster-admin --clusterrole=cluster-admin --serviceaccount=kube-system:default
-   kubectl --namespace kube-system create sa tiller
-   kubectl create clusterrolebinding tiller \
-       --clusterrole cluster-admin \
-       --serviceaccount=kube-system:tiller
-   helm init --service-account tiller
-   ```
+1. Install VirtualBox by [downloading and running an
+   installer](https://www.virtualbox.org/wiki/Downloads).
 
-9. Install / Upgrade JupyterHub Chart!
-   ```bash
-   helm upgrade --wait --install --namespace=hub hub jupyterhub/ -f minikube-config.yaml
-   ```
+1. Install Vagrant by [downloading and running an
+   installer](https://www.vagrantup.com/downloads.html).
 
-   You can easily change the options in `minikube-config.yaml` file to test what
-   you want, or create another `config.yaml` file & pass that as an additional
-   `-f config.yaml` file to the `helm upgrade` command.
+### Develop and run tests
 
-10. Retrieve the URL for your instance of JupyterHub:
+1. Start a prepared VM and SSH into it.
 
    ```bash
-   minikube service --namespace=hub proxy-public
-   ```
-
-   Navigate to the URL in your browser. You should now have JupyterHub running
-   on minikube.
+   ## if you have suspended a VM earlier, use "vagrat resume" instead
+   vagrant up
 
-11. Make the changes you want.
+   ## enter a SSH session with the VM
+   vagrant ssh
```

Comment:
I think we should avoid telling users to run any commands directly in the VM. All commands should ideally be run exclusively in the host environment. Having an ssh session open means folks lose all of the shell configuration they are used to, otherwise they lose history, completion, customization, etc.